backend/handler/auth/base_handler.py

```diff
@@ -165,16 +165,44 @@ class OpenIDHandler:
     RSA_ALGORITHM = "RS256"
 
     def __init__(self) -> None:
+        self._server_metadata: Optional[dict] = None
         self._rsa_key: Optional[RSAKey] = None
         self._rsa_key_lock = asyncio.Lock()
 
+    async def _fetch_server_metadata(self) -> dict:
+        """
+        Fetch the server metadata from the OIDC server
+        """
+        if self._server_metadata:
+            return self._server_metadata
+
+        server_metadata_url = (
+            f"{OIDC_SERVER_APPLICATION_URL}/.well-known/openid-configuration"
+        )
+        log.info("Fetching server metadata from %s", server_metadata_url)
+
+        httpx_client = ctx_httpx_client.get()
+        try:
+            response = await httpx_client.get(server_metadata_url, timeout=120)
+            response.raise_for_status()
+
+            self._server_metadata = response.json()
+            return response.json()
```

Comment:
We can avoid a second JSON unmarshal, by just returning `self._server_metadata` here.